ruby/tests/basic.rb

```diff
@@ -724,9 +756,12 @@ def test_dup
       m = TestMessage.new
       m.optional_string = "hello"
       m.optional_int32 = 42
-      m.repeated_msg.push TestMessage2.new(:foo => 100)
-      m.repeated_msg.push TestMessage2.new(:foo => 200)
-
+      tm1 = TestMessage2.new(:foo => 100)
+      tm2 = TestMessage2.new(:foo => 200)
+      m.repeated_msg.push tm1
+      assert m.repeated_msg[-1] == tm1
+      m.repeated_msg.push tm2
+      assert m.repeated_msg[-1] == tm2
```

Comment:
Tests fail (I assume you are saying just under Java/JRuby), really?

I was super-skeptical, because we check these things before cutting releases, and we haven't changed JRuby since. I just verified that tests pass with my checkout of the latest HEAD (`4b2a632`):

    [INFO] ------------------------------------------------------------------------
    [INFO] BUILD SUCCESS
    [INFO] ------------------------------------------------------------------------
    [INFO] Total time: 10.372 s
    [INFO] Finished at: 2015-05-02T18:52:21-07:00
    [INFO] Final Memory: 26M/222M
    [INFO] ------------------------------------------------------------------------
    Run options: 

    # Running tests:

    .............SS.......................

    Finished tests in 4.397000s, 8.6423 tests/s, 47828.5194 assertions/s.

      1) Skipped:
    test_json(BasicTest::MessageContainerTest) [/home/cfallin/.rvm/rubies/jruby-1.7.19/lib/ruby/1.9/minitest/unit.rb:949]:
    Unimplemented
    
      2) Skipped:
    test_json_maps(BasicTest::MessageContainerTest) [/home/cfallin/.rvm/rubies/jruby-1.7.19/lib/ruby/1.9/minitest/unit.rb:949]:
    Unimplemented
    
    38 tests, 210302 assertions, 0 failures, 0 errors, 2 skips